dissect/target/plugins/os/windows/_os.py

```diff
@@ -110,25 +112,137 @@ def ips(self) -> list[str]:
+            CurrentMinorVersionNumber sub key is not present or any other
+            registry error occurred.
+        """
+        version = None
         try:
-            r = self.target.registry.key(key)
-            product_name = r.value("ProductName").value
-            current_version = r.value("CurrentVersion").value
-            current_build_number = r.value("CurrentBuildNumber").value
-            return f"{product_name} (NT {current_version}) {current_build_number} {csd_version}"
+            reg_key = self.target.registry.key(self.CURRENT_VERSION_KEY)
+            version = reg_key.value("CurrentMinorVersionNumber").value
         except RegistryError:
             pass
 
+        return version
+
+    @internal
+    def nt_version(self) -> Optional[int]:
+        """Return the Windows NT version in x.y format.
+
+        For systems up to and including NT 6.3 (Win 8 / Win 2012 Server) this
+        will be the value of the CurrentVersion sub key.
+
+        For systems starting from NT 10.0 (Win 10) this will be a value
+        constructed from the combination of the CurrentMajorVersionNumber and
+        CurrentMinorVersionNumber sub keys.
+
+        Returns:
+            The string value of the NT version or ``None`` if the any of the
+            sub keys are not present or any other registry error occurred.
+        """
+        version = None
+
+        major_version = self.major_version()
+        if major_version is not None:
+            minor_version = self.minor_version()
+            if minor_version is None:
+                minor_version = ""
+            version = f"{major_version}.{minor_version}"
+        else:
+            version = self._legacy_current_version()
+
+        return version
+
+    @export(property=True)
+    def version(self) -> Optional[str]:
+        # Note that We don't show the "version" (aka FeatureRelease) as shown
+        # by WinVer.exe, whic uses the keys:
+        # ReleaseId: for Win 10 ReleaseId <= 2004
+        # DisplayVersion: for Win 10 ReleaseId >= 2009 (DisplayVersion = 20H2
+        #                 in this case)
+        #
+        # https://www.vcloudinfo.com/2020/12/how-to-decode-windows-version-numbers.html
+        version_string = None
+
+        try:
+            reg_key = self.target.registry.key(self.CURRENT_VERSION_KEY)
+
+            # Service Pack Version, when installed using the normal method (may not be present).
+            try:
+                csd_version = reg_key.value("CSDVersion").value
+            except RegistryError:
+                csd_version = ""
+
+            # Update Build Revision (seems to be present from NT 6.1 onwards).
+            try:
+                ubr = reg_key.value("UBR").value
+            except RegistryError:
+                ubr = ""
+
+            nt_version = self.nt_version()
+
+            # TODO: should we be this cautious or should we just try to make
```

Comment:
I don't understand the comment, what would that look like?

Reply:
Something like `<UNKNOWN Productname> (NT <UNKNONWN CurrentVersion>) <UNKNOWN BuidlVersion>` or `<UNKNOWN ProductName> (NT 6.3) <UNKNOWN CurrentBuildNumber>.1234`.  So all parts that can not be found, except for the parts that are ok to be missing (`CSDVersion` and `UBR`) wil be replaced with `<UNKNOWN $key>` .

For `nt_version` this will result in `<UNKNONWN CurrentVersion>` as `CurrentMajorVersionNumber` and `CurrentMinorVersionNumber` are allowed to not be present.

Reply:
Oh that might actually be pretty nice to do.